synapse/handlers/device.py

```diff
@@ -512,6 +512,14 @@ async def delete_devices(self, user_id: str, device_ids: List[str]) -> None:
                 user_id=user_id, device_id=device_id
             )
 
+            if self.hs.config.experimental.msc3890_enabled:
+                # Remove any local notification settings for this device in accordance
+                # with MSC3890.
+                await self.store.remove_account_data_for_user(
+                    user_id,
+                    f"org.matrix.msc3890.local_notification_settings.{device_id}",
+                )
```

Comment:
Hm, I've just noticed that we should use the [`AccountDataHandler.remove_account_data_for_user`](https://github.com/matrix-org/synapse/blob/c4456114e1a5471bb61cb45605e782263dc8233c/synapse/handlers/account_data.py#L215) method here instead, as that one will notify workers and Synapse modules that an update to account data has occurred.

Reply:
Fixed in 70d7cbf4f2796c84530c422a18d91e335fa8e766.